.github/workflows/scripts/scanner-fetch-nvd.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+base_url="https://services.nvd.nist.gov/rest/json/cves/2.0"
+# NVD CVE Api has limit of 2000 CVEs per page at maximum
+results_per_page=2000
+dir_name="nvd-data"
+
+# Function to fetch data for a given period
+fetch_data_for_period () {
+  local period_start_date=$1
+  local period_end_date=$2
+  local current_start_index=0
+  local total_results=0
+  local file_name
+  local success
+
+  while (( current_start_index < total_results || total_results == 0 )); do
+    file_name="${period_start_date}_${current_start_index}.json"
+    echo "Fetching data from ${period_start_date} to ${period_end_date}, starting at index ${current_start_index}..."
+
+    retry_intervals=(2 4 8) # Array of retry intervals in seconds
+    success=false
+
+    for i in "${retry_intervals[@]}"; do
+        if curl -s --fail --show-error -H "apiKey: $API_KEY" "$base_url?pubStartDate=$period_start_date&pubEndDate=$period_end_date&startIndex=$current_start_index" -o "$dir_name/$file_name"; then
+            success=true
+            break # Exit
+        else
+            sleep "$i" # Wait for next retry
+        fi
+    done
+
+    if [ "$success" = false ]; then
+        echo "Failed after retries"
+        exit 1
+    fi
+
+    if [ ! -s "$dir_name/$file_name" ] || ! jq empty "$dir_name/$file_name"; then
+      echo "Downloaded file is empty or invalid for the period: $period_start_date to $period_end_date, starting index $current_start_index"
+      exit 1
+    fi
+
+    total_results=$(jq '.totalResults' "$dir_name/$file_name") || { echo "jq command failed"; exit 1; }
+
+    current_start_index=$((current_start_index + results_per_page))
```

Comment:
I think we should consider using a dynamic `results_per_page` in case the API decides to ignore the original one. As in, once we get the data, update `results_per_page` to whatever the response says. But once we get to the next date range, we can reset the default to 2000. What do you think?